os/fs/smartfs/smartfs_smart.c

```diff
@@ -256,9 +254,9 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 			/* Don't truncate if open for APPEND */
 
 			if (!(oflags & O_APPEND)) {
-				/* Truncate the file as part of the open */
-
-				ret = smartfs_truncatefile(fs, &sf->entry, sf);
+				/* Truncate the file to length 0 as part of the open */
+				sf->filepos = 0;
```

Comment:
I think filepos will be changed in smartfs_shrinkfile, so this is unnecessary.

Reply:
shrinkfile saves sf->filepos at the start of the function.
So I am setting filepos to 0 as it is not set prior to this point.

Reply:
Yes smartfs_open will set filepos to 0 later.
But shrinkfile may hold a value of filepos that is uncertain,
I just wanted to avoid that.
Final call?